Automattic/jetpack
# Paid Newsletter: Selectively enable access panel

This PR goes with D107352-code.

Closes https://github.com/Automattic/jetpack/issues/30001

## Proposed changes:
* Respect the new filter `jetpack_subscriptions_newsletter_show_panel`
* Show "Add payments" if panel is showing and we have no plans configured

<img width="325" alt="Zrzut ekranu 2023-04-9 o 18 28 38" src="https://user-images.githubusercontent.com/3775068/230785289-d9a8eeba-35b8-477d-9b7c-fd0ab4bfbe34.png">


## Please take over that PR 

## Does this pull request change what data or activity we track or use?

No

## Testing instructions:
* Create a new site through newsletter flow `/setup/newsletter/intro`
* Sandbox public-api AND the site
* Apply D107352-code , sync this patch to wpcom.
* Confirm the panel is now showing and saying "Add payments"
* Click it, configure payments
* Panel should now show the access selector options
*
